### ISSUE TITLE
Inconsistent results in wday.numeric with and without labels

### ISSUE THREAD

**Issue by gregorp** [tidyverse/lubridate#1025]
Inspired by [this Stack Overflow question](https://stackoverflow.com/q/71254764/903061), the `wday.numeric` results seem out of sync as `week_start` changes depending on whether or not the `label` argument is TRUE or FALSE.  When `label = FALSE` changing `week_start` changes the output as expected. However when `label = TRUE` changing `week_start` has no effect. 

Reprex:

``` r
library(lubridate)
#> 
#> Attaching package: 'lubridate'
#> The following objects are masked from 'package:base':
#> 
#>     date, intersect, setdiff, union
wday(1, week_start = 1, label = FALSE)
#> [1] 7
wday(1, week_start = 2, label = FALSE)
#> [1] 6
wday(1, week_start = 3, label = FALSE)
#> [1] 5

wday(1, week_start = 1, label = TRUE)
#> [1] Sun
#> Levels: Mon < Tue < Wed < Thu < Fri < Sat < Sun
wday(1, week_start = 2, label = TRUE)
#> [1] Sun
#> Levels: Tue < Wed < Thu < Fri < Sat < Sun < Mon
wday(1, week_start = 3, label = TRUE)
#> [1] Sun
#> Levels: Wed < Thu < Fri < Sat < Sun < Mon < Tue
```

<sup>Created on 2022-02-24 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

<details style="margin-bottom:10px;">
<summary>
Session info
</summary>

``` r
sessionInfo()
#> R version 4.1.1 (2021-08-10)
#> Platform: aarch64-apple-darwin20 (64-bit)
#> Running under: macOS Big Sur 11.6
#> 
#> Matrix products: default
#> BLAS:   /Library/Frameworks/R.framework/Versions/4.1-arm64/Resources/lib/libRblas.0.dylib
#> LAPACK: /Library/Frameworks/R.framework/Versions/4.1-arm64/Resources/lib/libRlapack.dylib
#> 
#> locale:
#> [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
#> 
#> attached base packages:
#> [1] stats     graphics  grDevices utils     datasets  methods   base     
#> 
#> other attached packages:
#> [1] lubridate_1.7.10
#> 
#> loaded via a namespace (and not attached):
#>  [1] Rcpp_1.0.7      digest_0.6.28   withr_2.4.2     magrittr_2.0.1 
#>  [5] reprex_2.0.1    evaluate_0.14   highr_0.9       stringi_1.7.5  
#>  [9] rlang_0.4.11    cli_3.0.1       rstudioapi_0.13 fs_1.5.0       
#> [13] generics_0.1.0  rmarkdown_2.11  tools_4.1.1     stringr_1.4.0  
#> [17] glue_1.4.2      xfun_0.26       yaml_2.2.1      fastmap_1.1.0  
#> [21] compiler_4.1.1  htmltools_0.5.2 knitr_1.36
```
</details>

I think the issue is that the necessary shifts for `week_start` are [applied here](https://github.com/tidyverse/lubridate/blob/9cece5680d0280c5cd1ccdbc9309ccb4724b3a34/R/accessors-day.r#L82) in all cases, but in the labeled case the labels are [shifted in parallel](https://github.com/tidyverse/lubridate/blob/9cece5680d0280c5cd1ccdbc9309ccb4724b3a34/R/accessors-day.r#L92) effectively canceling them.

I **think** this could be fixed by making the numeric shift conditional on `label` being `FALSE`, that is, changing

```
## current version
  if (start != 7) {
    x <- 1 + (x + (6 - start)) %% 7
  }
```

to this:

```
## possible fix
  if (start != 7 & !label) {
    x <- 1 + (x + (6 - start)) %% 7
  }
```
Happy to do a bit more testing and submit a pull request if that would help out.

**Comment by vspinu** [comment]
I don't see any inconsistency. In both cases it's Sunday:

```R
> wday(1, week_start = 6, label = T)
[1] Sun
Levels: Sat < Sun < Mon < Tue < Wed < Thu < Fri
> wday(1, week_start = 6, label = F)
[1] 2
> as.integer(wday(1, week_start = 6, label = T))
[1] 2
```

**Comment by jaganmn** [comment]
`wday.numeric(x, week_start)` seems to assume that `x` uses the coding 1=Sunday, independently of `week_start`. `week_start` controls the coding of the result, and nothing else.

That's fine, but the coding that `x` is expected to follow ought to be documented, especially because it differs from the coding used for `week_start`, which is 1=Monday.

A more flexible `wday.numeric` would allow the user to specify the coding used by `x`.

**Comment by vspinu** [comment]
I share your concerns with you but I am afraid we are stuck with it for historical reasons. The current situation is neither ituitive nor seems to be very useful. 

If you have a concrete proposal of how to improve on the current interface I am all ears.

**Comment by gregorp** [comment]
Perhaps the solution of least resistance is updating the docs. As-is, they don't say anything about numeric input.

**Comment by jaganmn** [comment]
@vspinu I've added a PR with a couple of options for improving the interface. (Not sure if you would have have received a notification.)

**Comment by vspinu** [comment]
Yes. I saw. But I am still thinking about it. I don't quite like polluting the wday interface just because of one numeric method which arguably is not used that often. None of the proposal seems satisfactory to me. 

More I think about it more I am inclined to change the current state or do nothing. 

My main concern now is the inconsistency with numeric assignment:

```R
> tt <- ymd("2022-03-01")
> wday(tt, label = T)
[1] Tue
Levels: Sun < Mon < Tue < Wed < Thu < Fri < Sat
> wday(2, label = T, week_start = 2)
[1] Mon
Levels: Tue < Wed < Thu < Fri < Sat < Sun < Mon
> wday(tt, week_start = 2) <- 2
> wday(tt, label = T)
[1] Wed
Levels: Sun < Mon < Tue < Wed < Thu < Fri < Sat
```

I would expect the numeric wday to work consistently in both of these cases.

**Comment by vspinu** [comment]
The problem with the breaking change is that it's not trivial to get the old behavior. Can you think of a succint transition. In principle we can add a parameter to get the old behavior and obsolete it later if needed.
